onnxruntime/core/graph/model.cc

```diff
@@ -56,51 +55,47 @@ Model::Model(const std::string& graph_name,
   }
 
   for (const auto& domain : *p_domain_to_version) {
-    const gsl::not_null<OperatorSetIdProto*> opset_id_proto{model_proto_->add_opset_import()};
+    const gsl::not_null<OperatorSetIdProto*> opset_id_proto{model_proto_.add_opset_import()};
     opset_id_proto->set_domain(domain.first);
     opset_id_proto->set_version(domain.second);
   }
 
   std::unordered_map<std::string, const ONNX_NAMESPACE::FunctionProto*> model_functions_map;
   for (auto& func : model_functions) {
-    auto func_ptr = model_proto_->add_functions();
+    auto func_ptr = model_proto_.add_functions();
     func_ptr->CopyFrom(func);
     model_functions_map[func_ptr->name()] = func_ptr;
   }
 
   // need to call private ctor so can't use make_shared
   GSL_SUPPRESS(r .11)
-  graph_.reset(new Graph(model_proto_->mutable_graph(), *p_domain_to_version, IrVersion(), schema_registry,
+  graph_.reset(new Graph(model_proto_.mutable_graph(), *p_domain_to_version, IrVersion(), schema_registry,
                          logger, model_functions_map));
 }
 
 Model::Model(const ModelProto& model_proto, const IOnnxRuntimeOpSchemaRegistryList* local_registries,
              const logging::Logger& logger)
-    : Model(onnxruntime::make_unique<ModelProto>(model_proto), local_registries, logger) {
+    : Model(ModelProto(model_proto), local_registries, logger) {
```

Comment:
This will trigger copy ctor and hence should be avoided by the caller unless they don't want to pass on ownership of their ModelProto instance